### ISSUE TITLE
GEOSERVER admin password seems not be updated 

### ISSUE THREAD

**Issue by drumbsd** [GeoNode/geonode-project#209]
Hello guys. Trying to build a new environment using branch 3.3.x, I modified .env file, changing 

GEOSERVER_ADMIN_PASSWORD= geoserver

to this:

GEOSERVER_ADMIN_PASSWORD=4BfT9KAG9U

After that, I rebuilt the enviroment from scratch, but the geoserver password is still the same: Here steps to reproduce:

```
cd /home/geosolutions/geonode-project 
git pull
```
I'm currently using branch 3.3.x

```
export VIRTUALENVWRAPPER_PYTHON='/usr/bin/python3'
source /usr/local/bin/virtualenvwrapper.sh
mkvirtualenv --python=/usr/bin/python3 geonode_dev
pip install Django==2.2.15
django-admin startproject --template=./geonode-project -e py,sh,md,rst,json,yml,ini,env,sample,properties -n monitoring-cron -n Dockerfile geonode_dev
```
After that I modify .env and rebuild the environment

```
cd geonode_dev
docker-compose build --no-cache
docker-compose up -d --force-recreate
```

**Comment by t-book** [comment]
Please go with 3.2.x branch! Currently there is a lot of work going on which might bring you several problems with 3.3.x or master branch

**Comment by etj** [comment]
Hi @t-book , exactly, this issue is part of the work in progress to make geonode-project 3.3.x work! ;)

**Comment by giohappy** [comment]
~~@drumbsd I don't see where `GEOSERVER_ADMIN_PASSWORD` is ever used. AFAICS it's never passed down by docker-compose, and as you will know .env vars are only available at build stage, not down into the container.~~

~~The geonode entrypoint.sh invokes `geoserverfixture`, which in turn invokes [_geoserver_info_provision](https://github.com/GeoNode/geonode-project/blob/59f3906c785e51455e645abc5e364215957380ae/src/tasks.py?plain=1#L405). This method looks for `GEOSERVER_ADMIN_PASSWORD` to set the password, but I supsect that the env var won't be available.~~

~~I remember that in the past the entrypoint of geoserver-docker did some tricks to put the password inside the datadir, but I can't find this mechanism anymore.~~

@drumbsd forget my previous comment. The .env file should be correctly passed to the container service. I think we will need to put some logging inside the _geoserver_info_provision to check if the correct env var is parsed.

**Comment by drumbsd** [comment]
@giohappy some update:

Running **geoserverfixture** by hand I got a 401 Unauthorized.
So I changed the code in this way:

```
def _geoserver_info_provision(url):
    from django.conf import settings
    from geoserver.catalog import Catalog
    print("Setting GeoServer Admin Password...")
    cat = Catalog(
        url,
        #username=settings.OGC_SERVER_DEFAULT_USER,
        #password=settings.OGC_SERVER_DEFAULT_PASSWORD
        username='admin',
        password='geoserver'
```

And running the task manually I got a 200 Password has been updated. For some strange reason, looking with a recursive grep inside the code, I was not able to find neither OGC_SERVER_DEFAULT_USER nor OGC_SERVER_DEFAULT_PASSWORD.

I found this old issue [link ](https://github.com/GeoNode/geonode/issues/3175) where it seems that these 2 variables should be inside settings.py of geonode, but on settings.py under development.demo.geonode.org there's anything related to this. 

Any idea?

**Comment by giohappy** [comment]
@drumbsd this is odd, since genode-demo settings [should load](https://github.com/GeoNode/geonode-demo/blob/3.3.x/src/geonode_master/settings.py#L36) the default `geonode.settings.py` module, [which defines the two env vars](https://github.com/GeoNode/geonode/blob/3.3.x/geonode/settings.py#L992-L997).

I invite you to open the django shell (`python manage.py shell`) inside the django conainer:, and check the settings

```
from django.conf import settings
print(settings.OGC_SERVER_DEFAULT_USER)
```

**Comment by drumbsd** [comment]
@giohappy  Ok I rebuilt from scratch to have a clear situation. All dockers are up and running. The geoserver admin password is still "geoserver" even if:

```
root@c77f429ddaea:/usr/src/geonode_dev# python manage.py shell
Python 3.8.9 (default, Apr 10 2021, 15:47:22)
Type 'copyright', 'credits' or 'license' for more information
IPython 7.24.1 -- An enhanced Interactive Python. Type '?' for help.

In [1]: from django.conf import settings
   ...: print(settings.OGC_SERVER_DEFAULT_USER)
admin

In [2]: from django.conf import settings
   ...: print(settings.OGC_SERVER_DEFAULT_PASSWORD
   ...: )
4BfT9KAG9UWe5PUa

In [3]:
Do you really want to exit ([y]/n)? y
root@c77f429ddaea:/usr/src/geonode_dev# invoke geoserverfixture
********************geoserver fixture********************************
Setting GeoServer Admin Password...
Response Code: 401
WARNING: GeoServer admin password *NOT* updated: code [401]
```

And the .env

```
GEOSERVER_ADMIN_PASSWORD=4BfT9KAG9UWe5PUa
```

So basically seems that settings OGC_SERVER_DEFAULT_USER and OGC_SERVER_DEFAULT_PASSWORD are correctly configured.

Maybe the problem is this:
* Geoserver start with default password "geoserver" in all cases.
* Fixture try to modify it using OGC_SERVER_DEFAULT_PASSWORD that is not the current admin password of the geoserver
* It got a 401 response.

That's because to properly configure the new password we need to connect to geoserver with the default password.

Maybe I'm wrong but is the only reason I was able to find. Maybe something like this:

* Connect with OGC_SERVER_DEFAULT_PASSWORD
* If got 200 , then OK
* else try to use "geoserver" password

**Comment by giohappy** [comment]
The issue has been spot: the `_geoserver_info_provision` doesn't work and it's conceptually wrong!

1 - as @drumbsd notice `OGC_SERVER_DEFAULT_PASSWORD` [is set to](https://github.com/GeoNode/geonode/blob/3.2.x/geonode/settings.py#L997) `GEOSERVER_ADMIN_PASSWORD` if this env var is set. So itr tries to connect to Geoserver with a password yet to be set, and it fails of course
2 - Even if `OGC_SERVER_DEFAULT_PASSWORD` was using the initial defauit password ("geoserver") the method would work only for the first time if the password hasn't been changed on Geoserver. However it will fail the next times.

Long story short, this method is wrong and misleading. I don't see any simple and robust solution, if we want to keep using the GeoServer REST API to set its password. IMHO the method, the `geoserverfixture` task and it's invocation should be deleted. 
Opinions @randomorder?

**Comment by giohappy** [comment]
Probably GeoNdoe itself will have to be updated https://github.com/GeoNode/geonode/issues/7535
